### ISSUE TITLE
Bad result on my own fisheye dataset with nerfacto method.

### ISSUE THREAD

**Issue by qhdqhd** [nerfstudio-project/nerfstudio#1223]
My dataset has about 400 fisheye images with a resolution of 1920x1080.
![0018](https://user-images.githubusercontent.com/42370719/212005928-194253dd-a618-43df-8769-90d85b1e58d7.png)
![0200](https://user-images.githubusercontent.com/42370719/212006741-ebb82249-ef12-4fd7-a7a7-020907fafb13.png)
![Uploading 0031.png…]()
![0076](https://user-images.githubusercontent.com/42370719/212006801-e6c539ac-eb61-4dd2-b36f-8b11f0610fc9.png)
![0184](https://user-images.githubusercontent.com/42370719/212006816-1f622632-72a4-4f10-8660-cd9c4ce3a978.png)

Camera poses are calibrated by COLMAP.
![ZE%3)G{9L@LFS{QNF1A`U~D](https://user-images.githubusercontent.com/42370719/212006702-04bcb1f8-43b4-4654-8c54-7074c5b0c7dd.png)
However, the rendered image is very blurred and there are many floating objects.
![MAAE_ZLU~3$X93CXL(_@R)V](https://user-images.githubusercontent.com/42370719/212038383-d7d8e318-3982-4cd2-95e6-819fbc2d9442.png)

https://user-images.githubusercontent.com/42370719/212004975-657f83f3-e6cf-43f0-9a44-114f760b0178.mp4

I have ruled out the problem of image quantity and image resolution. Although camera calibration is not perfect. I find CUDA version instant-ngp by NVIDIA can get better results than nerfacto, although not perfect. So camera calibration may be not the key problem.
Following is instant-ngp results by NVIDIA.
![6)VJ$NXM_6U4G0%MT ZRF1D](https://user-images.githubusercontent.com/42370719/212010076-329f92a0-8aa2-40a8-b56f-a39c15faf20d.png)

![G0GW)2 LOODHBI 660ZWWV5](https://user-images.githubusercontent.com/42370719/212007191-e19b0e48-edcd-463e-a281-a4dc1b331ead.png)

So I wonder why?

I compare my dataset with your public Giannini-Hall dataset. Both datasets shoot outdoor scenes, but our camera's space movement range is small, so the scene box is small and can't cover the scene. Is this the main reason? How can I solve this problem?
![~)~JR{%YUJM6(PAE9DW(6WL](https://user-images.githubusercontent.com/42370719/212009549-06fb9a04-1803-4070-9f2d-cade3c8f7a93.png)

**Comment by tancik** [comment]
How did you compute the poses? Did you run `ns-process-data images --camera-type fisheye ...`

**Comment by qhdqhd** [comment]
> 你是如何计算姿势的？你跑了吗`ns-process-data images --camera-type fisheye ...` “场景框”的定义在nerfacto和instant-ngp中有点不同。对于 nerfacto，我们希望盒子只包含训练视图，而不是整个场景。

I just calculate camera pose with colmap (opencv-fisheye), and I get transform.json file by a script (https://github.com/NVlabs/instant-ngp/blob/master/scripts/colmap2nerf.py) in instant-ngp.

**Comment by tancik** [comment]
Gotcha, so that script doesn't set the `camera_model` param in the `transforms.json` (https://docs.nerf.studio/en/latest/quickstart/data_conventions.html#camera-intrinsics)
In this case it should be `OPENCV_FISHEYE`

I'd recommend using the `ns-process-data` script.

**Comment by qhdqhd** [comment]
Thank you, but I have set the camera_model by imitating your official data (Giannini-Hall) already.
This is my transform.json.
![EMG)6M`A(_JN_X6@BNPU@WI](https://user-images.githubusercontent.com/42370719/213059565-3bd52506-b2d3-4e83-ab4f-ff668dd14b38.png)

**Comment by tancik** [comment]
Are the results also bad if you use `ns-process-data`?

**Comment by qhdqhd** [comment]
> Are the results also bad if you use `ns-process-data`?

I have not tried it, I will test it. Thank you.

**Comment by matsuren** [comment]
> Thank you, but I have set the camera_model by imitating your official data (Giannini-Hall) already. This is my transform.json. ![EMG)6M`A(_JN_X6@BNPU@WI](https://user-images.githubusercontent.com/42370719/213059565-3bd52506-b2d3-4e83-ab4f-ff668dd14b38.png)

@qhdqhd 
There is a bug when you put distortion parameters inside frames in json file, which should be fixed by #1268 PR.
I'm not familiar with instant-ngp convert script, but that might be a cause of problems.

**Comment by qhdqhd** [comment]
> 

I really help,thanks!

**Comment by qhdqhd** [comment]
> > 谢谢，但我已经通过模拟您的官方数据（Giannini-Hall）设置了camera_model。这是我的transform.json。![EMG)6M`A(_JN_X6@BNPU@WI](https://user-images.githubusercontent.com/42370719/213059565-3bd52506-b2d3-4e83-ab4f-ff668dd14b38.png)
> 
> @qhdqhd 在json文件中将丢失真参数据放入控制台时会出现错误，该错误应由[#](https://github.com/nerfstudio-project/nerfstudio/pull/1268) 1​​268 PR修复。 我不熟悉instant-ngp转换脚本，但这可能是问题的原因。

It really helps!

**Comment by rockywind** [comment]
Hi, @tancik 
The pictures showed in the view is like a circle, but the car war move forward.
The training result is bad.
The training set number is 270. validation set number is 30.
This is the data process scripts.
````
ns-process-data images --data /DATA2/xiaoquan.wang/nerfstudio/data/own_data/data_fish_23_18/image_A_640_480/ --output_dir  /DATA2/xiaoquan.wang/nerfstudio/data/own_data/data_fish_23_18/ --camera-type fisheye --matching-method exhaustive --num-downscales 1 --sfm-tool hloc
````
This is the training script .
````
export CUDA_VISIBLE_DEVICES=2
ns-train nerfacto  --viewer.websocket-port=7009 \
                --experiment-name data_fish_23_18_300 \
                --data data/own_data/data_fish_23_18_300 \
                --vis tensorboard 
````
![image](https://user-images.githubusercontent.com/14838781/217741283-2e2011d6-6ed4-4388-8334-3fdf0dda95fe.png)
This is the training log.
![image](https://user-images.githubusercontent.com/14838781/217742219-158fe03d-a9a7-4ad0-8e6c-5cc83439c025.png)
![image](https://user-images.githubusercontent.com/14838781/217742631-f853bac9-53b0-40d9-8fd8-5e2c85dc7abf.png)
![image](https://user-images.githubusercontent.com/14838781/217742700-a9779710-1338-4f2e-9bd6-fd53eb7d19e7.png)
![image](https://user-images.githubusercontent.com/14838781/217742744-052c51d4-0bb4-49e0-bb53-3e5a9c2390ea.png)
![image](https://user-images.githubusercontent.com/14838781/217742874-5c39bcba-897d-46a1-8630-20ab9ec5b82a.png)

**Comment by tancik** [comment]
It looks like one of the training images has a bad pose. This is messing up the auto scaling and thus making the scene very small which ruins the reconstruction. Ideally you remove this bad image from the training set. Alternatively add `nerfstudio-data --auto-scale-poses False --scale-factor X` and try a few different values for `X` where the goal is to get the main clump of images to roughly fit the bounding box like the following screenshot
<img width="783" alt="image" src="https://user-images.githubusercontent.com/3310961/217745332-295d7270-637e-4ced-b54b-01fac646641f.png">

**Comment by rockywind** [comment]
Hi,
This is the data process log.
````
2023/02/09 19:00:33 hloc INFO] Finished exporting matches.
[2023/02/09 19:00:33 hloc INFO] Creating an empty database...
[2023/02/09 19:00:33 hloc INFO] Importing images into the database...
[2023/02/09 19:00:43 hloc INFO] Importing features into the database...
100%|██████████████████████████████████████████| 301/301 [00:00<00:00, 596.52it/s]
[2023/02/09 19:00:43 hloc INFO] Importing matches into the database...
100%|██████████████████████████████████████| 45150/45150 [01:42<00:00, 439.00it/s]
[2023/02/09 19:02:29 hloc INFO] Performing geometric verification of the matches...
[2023/02/09 19:09:40 hloc INFO] Running 3D reconstruction...
WARNING: Logging before InitGoogleLogging() is written to STDERR
W0209 19:21:15.279369 3997066 levenberg_marquardt_strategy.cc:116] Linear solver failure. Failed to compute a step: Eigen failure. Unable to perform dense Cholesky factorization.
W0209 19:21:15.280140 3997066 levenberg_marquardt_strategy.cc:116] Linear solver failure. Failed to compute a step: Eigen failure. Unable to perform dense Cholesky factorization.
W0209 19:21:15.288179 3997066 levenberg_marquardt_strategy.cc:116] Linear solver failure. Failed to compute a step: Eigen failure. Unable to perform dense Cholesky factorization.
W0209 19:21:15.296222 3997066 levenberg_marquardt_strategy.cc:116] Linear solver failure. Failed to compute a step: Eigen failure. Unable to perform dense Cholesky factorization.
W0209 19:23:03.622383 3997066 levenberg_marquardt_strategy.cc:116] Linear solver failure. Failed to compute a step: Eigen failure. Unable to perform dense Cholesky factorization.
W0209 19:23:03.642295 3997066 levenberg_marquardt_strategy.cc:116] Linear solver failure. Failed to compute a step: Eigen failure. Unable to perform dense Cholesky factorization.
W0209 19:23:03.644703 3997066 levenberg_marquardt_strategy.cc:116] Linear solver failure. Failed to compute a step: Eigen failure. Unable to perform dense Cholesky factorization.
````
Why there are so many ``Linear solver failure. Failed to compute a step: Eigen failure. Unable to perform dense Cholesky factorization.``

**Comment by tancik** [comment]
No idea. These also look like warnings, not necissarily a problem. Are you using hloc instead of colmap? Does colmap fail?

**Comment by rockywind** [comment]
Hi,
This is the data process script.
````
ns-process-data images \
--data /DATA2/xiaoquan.wang/nerfstudio/data/own_data/data_fish_23_18_300/source/ \
--output_dir  /DATA2/xiaoquan.wang/nerfstudio/data/own_data/data_fish_23_18_300/ \
--camera-type fisheye \
--matching-method exhaustive \
--num-downscales 1
````
This is the error log.
````
( ●    ) Copying images...[10:04:52] 🎉 Done copying images.                                                             process_data_utils.py:191
[10:05:26] 🎉 Done downscaling images.                                                         process_data_utils.py:303
🌓  Running COLMAP feature extractor...──────────────────────────────────────────────  💀 💀 💀 ERROR 💀 💀 💀  ───────────────────────────────────────────────
Error running command: colmap feature_extractor --database_path 
/DATA2/xiaoquan.wang/nerfstudio/data/own_data/data_fish_23_18_300/colmap/database.db --image_path 
/DATA2/xiaoquan.wang/nerfstudio/data/own_data/data_fish_23_18_300/images --ImageReader.single_camera 1 
--ImageReader.camera_model OPENCV_FISHEYE --SiftExtraction.use_gpu 1
────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
qt.qpa.xcb: could not connect to display 
qt.qpa.plugin: Could not load the Qt platform plugin "xcb" in 
"/home/zongmu/anaconda3/envs/nerfstudio/lib/python3.8/site-packages/cv2/qt/plugins" even though it was found.
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may 
fix this problem.

Available platform plugins are: eglfs, linuxfb, minimal, minimalegl, offscreen, vnc, xcb.

*** Aborted at 1675994736 (unix time) try "date -d @1675994736" if you are using GNU date ***
PC: @     0x7f798609e00b gsignal
*** SIGABRT (@0x3e800017b0f) received by PID 97039 (TID 0x7f798189a900) from PID 97039; stack trace: ***
    @     0x7f7987c8c631 (unknown)
    @     0x7f79871cf420 (unknown)
    @     0x7f798609e00b gsignal
    @     0x7f798607d859 abort
    @     0x7f7986662aad QMessageLogger::fatal()
    @     0x7f7986c447ae QGuiApplicationPrivate::createPlatformIntegration()
    @     0x7f7986c45708 QGuiApplicationPrivate::createEventDispatcher()
    @     0x7f7986869f55 QCoreApplicationPrivate::init()
    @     0x7f7986c47543 QGuiApplicationPrivate::init()
    @     0x7f798734f3bd QApplicationPrivate::init()
    @     0x564005d34602 RunFeatureExtractor()
    @     0x564005d20eaf main
    @     0x7f798607f083 __libc_start_main
    @     0x564005d24f6e _start
Aborted (core dumped)
````
I run the data process script in a server.
I think it need a monitor, when using colmap.

**Comment by tancik** [comment]
You can try adding `--no-gpu` to the command.

**Comment by rockywind** [comment]
Thanks, it works.
